docs/howto/https/index.adoc

```diff
@@ -0,0 +1,64 @@
+
+[#enabling-https_{context}]
+= Enabling HTTPS
+
+{Thorntail} provides several ways of enabling HTTPS, depending on your use case.
+Here, we describe the most common and most straightforward way.
+
+.Prerequisites
+
+* Keystore in the JKS format. This keystore must include a private key and a corresponding certificate.
++
+--
+If you don't have a keystore, you can create one using `keytool`:
+
+[source,bash,options="nowrap"]
+----
+keytool -genkeypair -keystore my-keystore.jks -storepass my-password -alias my-https-key -keypass my-password -dname CN=localhost -sigalg SHA1withRSA -validity 100 -keyalg RSA -keysize 2048
+----
+
+This keystore is only suitable for testing, as the certificate is self-signed.
+
+To obtain a production-ready certificate, trusted by web browsers, you can use link:https://letsencrypt.org/[Let's Encrypt] or any other certificate authority.
+--
+
+.Procedure
+
+. Include the `management` fraction in the `pom.xml` file in your application:
++
+.pom.xml
+[source,xml]
+----
+<dependencies>
+include::pom.xml[tag=management-fraction,indent=2]
+</dependencies>
+----
+
+. Add the HTTPS properties to your application configuration:
++
+.src/main/resources/project-defaults.yml
+[source,xml]
+----
+include::src/main/resources/project-defaults.yml[]
+----
+
+. Start the application and use `curl` to verify that the application is accessible over HTTPS:
++
+[source,bash,options="nowrap"]
+----
+curl -k -v https://localhost:8443/
+----
+
+.Additional resources
+
+For more examples of how to enable HTTPS, see the link:https://github.com/rhoar-qe/thorntail-test-suite/tree/master/protocols[Thorntail test suite^].
+It includes:
+
+* 1-way TLS
+** Note that this is simply a technical term for what we've just done, setting up an HTTPS server.
+   Thorntail supports several ways of setting up 1-way TLS, so there are multiple examples.
+   They are all functionally equivalent.
```

Comment:
{Thorntail} supports several ways of setting up 1-way TLS, so there are multiple examples.

**Questions**
- Should we change 1-way, 2-way TLS to:
One-way TLS, Two-way TLS

- Do we need to add the other content added in the note? 1-way TLS, 2-way TLS are known terms?
https://docs.apigee.com/api-platform/system-administration/about-ssl

Reply:
Good point re "one-way" instead of "1-way", I'll do that.

A link to some article would be great.